server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailQuery.scala

```diff
@@ -32,6 +32,14 @@ case class UnsupportedSortException(unsupportedSort: String) extends Unsupported
 case class UnsupportedFilterException(unsupportedFilter: String) extends UnsupportedOperationException
 case class UnsupportedRequestParameterException(unsupportedParam: String) extends UnsupportedOperationException
 
+sealed trait FilterQuery
+
+sealed trait Operator
+case object And extends Operator
+
+case class FilterOperator(operator: Operator,
+                          conditions: Seq[FilterCondition]) extends FilterQuery
```

Comment:
needs to be `conditions: Seq[FilterQuery]` to support nested operators (add a test about this too?)